mlrun/api/schemas/artifact.py

```diff
@@ -43,6 +45,20 @@ def to_kinds_filter(self) -> typing.Tuple[typing.List[str], bool]:
             )
 
 
+class ArtifactObject(pydantic.BaseModel):
```

Comment:
I don't think this should be `ArtifactObject`, it should only be an identifier for an Artifact, so would call it `ArtifactIdentifier` and only hold fields which are identifying an Artifact, so I would remove the `labels` and `tag` from the list here, since they are just descriptions for an object and may be changed for the same object. This leaves us with `name` (which should be `key`), `uid` which is missing and `iter`, and when we add content-hash then this as well.
As for `kind` - just add a comment that this is the artifact kind, not generic object kind.

Reply:
added uid support